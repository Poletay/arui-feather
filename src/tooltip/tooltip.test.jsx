import React from 'react';
import { mount } from 'enzyme';

import Tooltip, { combine } from './tooltip';

describe('tooltip', () => {
    it('should render without problems', () => {
        let elPopup = mount(<Tooltip uniqID='1'><span>TEST</span></Tooltip>);

        expect(elPopup).toMatchSnapshot();
    });

    it('should combine correct selector', () => {
        expect(
            combine(['desktop', 'direction', 'sub-direction'], 'popup')
        ).toEqual('.popup_desktop.popup_direction.popup_sub-direction');
    });
});
