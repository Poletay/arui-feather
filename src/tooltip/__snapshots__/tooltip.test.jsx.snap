// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip should render without problems 1`] = `
<Tooltip
  direction="bottom"
  forceDirection={false}
  trigger="click"
  uniqID="1"
>
  <div
    className="tooltip tooltip_theme_alfa-on-white"
    id="1"
  >
    <Link
      checked={false}
      className="tooltip__children"
      disabled={false}
      iconPosition="left"
      onClick={[Function]}
      pseudo={true}
      size="m"
      tabIndex={0}
      url="#"
    >
      <a
        className="link link_pseudo link_size_m link_theme_alfa-on-white tooltip__children"
        href="#"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        tabIndex={0}
      >
        <span>
          TEST
        </span>
      </a>
    </Link>
    <Mq
      query="--tablet-s"
    />
    <Mq
      query="--mobile"
    />
  </div>
</Tooltip>
`;
